src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2278,16 +2283,15 @@ private bool RefMustBeObeyed(bool isDelegateCreation, ArgumentSyntax argumentSyn
                 {
                     case BoundKind.PropertyAccess:
                     case BoundKind.IndexerAccess:
-                        return CheckIsVariable(argumentSyntax, arg, BindValueKind.RefOrOut, false, diagnostics);
+                        CheckIsVariable(argumentSyntax, arg, BindValueKind.RefOrOut, false, diagnostics);
+                        return;
```

Comment:
There is still a problem here. I should have done `|=` instead. Also `CheckIsVariable` returns `true` for success, not for error. I will try to find a scenario to hit this.